network/wsNetwork.go

```diff
@@ -683,6 +686,11 @@ func (wn *WebsocketNetwork) GetPeers(options ...PeerOption) []Peer {
 	return outPeers
 }
 
+// ExtendPeerList is used for testing the node w/ manual phonebook configurations
+func (wn *WebsocketNetwork) ExtendPeerList(peers ...string) {
```

Comment:
Is this absolutely necessary? 
I am sure this interface was eliminated for a good reason, and returning it here is not a good idea.
Also, it is desirable to use the pattern that the production code uses for a more realistic test. 

If it turns out this is absolutely necessary, then instead of adding this function here, it is appropriate to create a test directory, just like `go-algorand/ledger/testing`, and add this function there. 
Leaving this here, one day the note can be overlooked, and bad things happen.

Reply:
I agree with Shant lets not extend the interface. In the test it is OK to cast gossip node to `*WebsocketNetwork`.
Another thing to consider - `ph.ExtendPeerList` is not even used inside network package, only in couple tests, and the real function is `ReplacePeerList` that is called from exported `NewWebsocketNetwork`.
Since `startAndConnectNodes` (and so that `connectPeers`) is used only in test setup, so tests setup code might modify node's network with `NewWebsocketNetwork` as needed.

Reply:
Yep, I ended up removing this by manually setting the localhost port of each node being started and passing them into `MakeFull` phonebook addresses. That way they all know about the other nodes since the ports aren't being randomized after startup.